packages/office-ui-fabric-react/src/components/DetailsList/DetailsColumn.tsx

```diff
@@ -0,0 +1,255 @@
+import * as React from 'react';
+import * as stylesImport from './DetailsHeader.scss';
+const styles: any = stylesImport;
+import { Icon } from '../../Icon';
+import { BaseComponent, css, IRenderFunction, createRef, IDisposable } from '../../Utilities';
+import { IColumn, ColumnActionsMode } from './DetailsList.types';
+
+import { ITooltipHostProps } from '../../Tooltip';
+import { IDragDropHelper, IDragDropOptions } from './../../utilities/dragdrop/interfaces';
+
+const INNER_PADDING = 16; // Account for padding around the cell.
+const ISPADDED_WIDTH = 24;
+const MOUSEDOWN_PRIMARY_BUTTON = 0; // for mouse down event we are using ev.button property, 0 means left button
+
+export interface IDetailsColumnProps extends React.Props<DetailsColumn> {
+  componentRef?: () => void;
+  column: IColumn;
+  columnIndex: number;
+  parentId?: string;
+  onRenderColumnHeaderTooltip?: IRenderFunction<ITooltipHostProps>;
+  onColumnClick?: (ev: React.MouseEvent<HTMLElement>, column: IColumn) => void;
+  onColumnContextMenu?: (column: IColumn, ev: React.MouseEvent<HTMLElement>) => void;
+  dragDropHelper?: IDragDropHelper | null;
+  isDraggable?: boolean;
+  setDraggedItemIndex?: (itemIndex: number) => void;
+  isDropped?: boolean;
+}
+
+export class DetailsColumn extends BaseComponent<IDetailsColumnProps> {
+  private _root: any;
+  private _dragDropSubscription: IDisposable;
+
+  constructor(props: IDetailsColumnProps) {
+    super(props);
+
+    this._root = createRef();
+    this._onDragStart = this._onDragStart.bind(this);
+    this._onDragEnd = this._onDragEnd.bind(this);
+    this._onRootMouseDown = this._onRootMouseDown.bind(this);
+  }
+
+  public render() {
+    const { column, columnIndex, parentId, isDraggable } = this.props;
+    const { onRenderColumnHeaderTooltip = this._onRenderColumnHeaderTooltip } = this.props;
+
+    return [
+      <div
+        key={column.key}
+        ref={this._root}
+        role={'columnheader'}
+        aria-sort={column.isSorted ? (column.isSortedDescending ? 'descending' : 'ascending') : 'none'}
+        aria-disabled={column.columnActionsMode === ColumnActionsMode.disabled}
+        aria-colindex={columnIndex}
+        className={css(
+          'ms-DetailsHeader-cell',
+          styles.cell,
+          column.headerClassName,
+          column.columnActionsMode !== ColumnActionsMode.disabled && 'is-actionable ' + styles.cellIsActionable,
+          !column.name && 'is-empty ' + styles.cellIsEmpty,
+          (column.isSorted || column.isGrouped || column.isFiltered) && 'is-icon-visible',
+          column.isPadded && styles.cellWrapperPadded
+        )}
+        data-is-draggable={isDraggable}
+        draggable={isDraggable}
```

Comment:
Why do we need both data-is-draggable and draggable attributes?

Reply:
I have followed DetailsRow, may be kept both for legacy reasons,

Reply:
@yiminwu @ThomasMichon - do any of you recall why we need both these attributes? Seems sort of redundant for the new component.

Reply:
I didn't remember.

---
In reply to: [196860743](https://github.com/OfficeDev/office-ui-fabric-react/pull/4857#discussion_r196860743) [](ancestors = 196860743)